polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -96,6 +96,12 @@ ValueCategory ValueCategory::getNullValue(OpBuilder &Builder, Location Loc,
   return {ZeroVal, false};
 }
 
+ValueCategory ValueCategory::getUndefValue(OpBuilder &Builder, Location Loc,
+                                           Type Type) {
+  // TODO: Replace with higher-level undef operation when defined.
```

Comment:
There are already plans for adding this. [See](https://mlir.llvm.org/OpenMeetings/2022-11-03-Poison-Semantics-For-MLIR.pdf)